data-prepper-plugins/avro-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/avro/AvroInputCodecTest.java

```diff
@@ -0,0 +1,204 @@
+    private static final String FILE_SUFFIX = ".avro";
+    private static final String INVALID_AVRO_INPUT_STREAM = "Invalid Avro Input Stream";
+
+    @Mock
+    private Consumer<Record<Event>> eventConsumer;
+
+    private AvroInputCodec avroInputCodec;
+
+    private static FileInputStream fileInputStream;
+
+    @TempDir
+    private static java.nio.file.Path path;
+
+    private AvroInputCodec createObjectUnderTest(){
+        return new AvroInputCodec();
+    }
+
+    @BeforeEach
+    void setup(){
+        avroInputCodec=createObjectUnderTest();
+    }
+
+    @Test
+    public void test_when_nullInputStream_then_throwsException(){
+        avroInputCodec=new AvroInputCodec();
+        Consumer<Record<Event>> eventConsumer = mock(Consumer.class);
+        assertThrows(NullPointerException.class,()->
+                avroInputCodec.parse(null, eventConsumer));
+
+        verifyNoInteractions(eventConsumer);
+
+    }
+
+    @Test
+    public void parse_with_Invalid_InputStream_then_catches_exception()  {
+        avroInputCodec=new AvroInputCodec();
+        Consumer<Record<Event>> eventConsumer = mock(Consumer.class);
+        Assertions.assertDoesNotThrow(()->
+                avroInputCodec.parse(createInvalidAvroStream(), eventConsumer));
+    }
+
+    @Test
+    public void parse_with_null_Consumer_throws() {
+        avroInputCodec=createObjectUnderTest();
+
+        final InputStream inputStream = mock(InputStream.class);
+        assertThrows(NullPointerException.class,()->
+                avroInputCodec.parse(inputStream,null));
+
+        verifyNoInteractions(inputStream);
+
+    }
+
+    @Test
+    public void parse_with_empty_InputStream_does_not_call_Consumer() throws IOException{
+        avroInputCodec=createObjectUnderTest();
+        Consumer<Record<Event>> eventConsumer = mock(Consumer.class);
+        final ByteArrayInputStream emptyInputStream=new ByteArrayInputStream(new byte[]{});
+        avroInputCodec.parse(emptyInputStream, eventConsumer);
+        verifyNoInteractions(eventConsumer);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints={1,10,100,1000})
+    public void test_HappyCaseAvroInputStream_then_callsConsumerWithParsedEvents(final int numberOfRecords)throws IOException{
+
+        avroInputCodec=createObjectUnderTest();
+        InputStream inputStream=createRandomAvroStream(numberOfRecords);
+
+        avroInputCodec.parse(inputStream,eventConsumer);
+
+        final ArgumentCaptor<Record<Event>> recordArgumentCaptor = ArgumentCaptor.forClass(Record.class);
+        verify(eventConsumer, times(numberOfRecords)).accept(recordArgumentCaptor.capture());
+        final List<Record<Event>> actualRecords = recordArgumentCaptor.getAllValues();
+        assertThat(actualRecords.size(), equalTo(numberOfRecords));
+
+        for (final Record<Event> actualRecord : actualRecords) {
+
+            assertThat(actualRecord, notNullValue());
+            assertThat(actualRecord.getData(), notNullValue());
```

Comment:
Please verify that the events in the records have `person`, `age` and `name` are present and have expected values.